eth/client.go

```diff
@@ -418,7 +419,161 @@ func (c *client) Bond(amount *big.Int, to ethcommon.Address) (*types.Transaction
 		}
 	}
 
-	return c.BondingManagerSession.Bond(amount, to)
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	// Get delegator
+	delegator, err := c.GetDelegator(sender)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get delegator")
+	}
+
+	isFull := int64(len(transcoders)) == maxSize.Int64()
+
+	// Switching delegate's calculate old delegate positions
+	var oldHints lpTypes.TranscoderPoolHints
+	if delegator.DelegateAddress != to {
+		// Get total bonded
+		totalBonded, err := c.TranscoderTotalStake(delegator.DelegateAddress)
+		if err != nil {
+			return nil, err
+		}
+		oldHints = simulateTranscoderPoolUpdate(delegator.DelegateAddress, new(big.Int).Sub(totalBonded, delegator.BondedAmount), transcoders, isFull)
+	}
+
+	// Get total bonded
+	totalBonded, err := c.TranscoderTotalStake(to)
+	if err != nil {
+		return nil, err
+	}
+	newStake := totalBonded.Add(totalBonded, amount)
```

Comment:
If the delegator is switching delegates we also need to add the delegator's pending stake here since that will be added to the new delegate's stake as well.

Reply:
Bump. Looks like now we're including the delegator's pending stake, but not including the additional amount if it is non-zero (i.e. if new LPT is staked).